### PR TITLE
OKTA-452297 - Updates to OIE SDK Guides for Node.js re authenticatorKey

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** Updated the OIE SDK Guides for Node.js language selector updating (and adding) details on the `authenticatorKey`
- **Is this PR related to a Monolith release?** no

### Resolves:

* [OKTA-452597](https://oktainc.atlassian.net/browse/OKTA-452597)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-sign-in-pwd-email/main/nodejs/integrationsteps.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ Build a sign-in page that captures both the user’s name and password, as shown
 
 When the user initiates the sign-in process, your app needs to create a new `OktaAuth` object and set its `username` and `password` properties to the values entered by the user. Send this object to the [`idx.authenticate`](https://github.com/okta/okta-auth-js/blob/master/docs/idx.md#idxauthenticate) method to start the authentication process. This call begins the multi-factor authentication flow.
 
-If the username and password are valid, `idx.authenticate` returns a status of `Idx.Status:PENDING`. This status indicates that an additional factor needs to be verified before the sign in. In addition to the status, the `nextStep` field that is included in the response identifies the input parameters of the next step, which in this case is for the email authenticator:
+If the username and password are valid, `idx.authenticate` returns a status of `Idx.Status:PENDING`. This status indicates that an additional factor needs to be verified before the sign in. In addition to the status, the `nextStep` field that is included in the response identifies the input parameters of the next step, which in this case is for the email authenticator key:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-sign-in-pwd-email/main/nodejs/integrationsteps.md:13]:
```suggestion
If the username and password are valid, `idx.authenticate` returns a status of `Idx.Status:PENDING`. This status indicates that an additional factor needs to be verified before the user signs in. In addition to the status, the `nextStep` field that is included in the response identifies the input parameters of the next step, which in this case is for the email authenticator key:
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-sign-in-pwd-phone/main/nodejs/integrationsteps.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ Build a sign-in page that captures the username and password, as shown in the fo
 
 When the user initiates the sign-in process, your app needs to create a new `OktaAuth` object and set its `username` and `password` properties to the values entered by the user. Send this object to `idx.authenticate` to start the authentication process. See [idx.Authenticate](https://github.com/okta/okta-auth-js/blob/master/docs/idx.md#idxauthenticate) for more information. This call begins the multifactor authentication flow.
 
-If the username and password are valid, `idx.authenticate` returns a status of `Idx.Status:PENDING`. This status indicates that an additional factor needs to be verified before the user is successfully signed in. The response also includes a `nextStep` field that identifies the input parameters of the next step, in this case an authenticator of type phone:
+If the username and password are valid, `idx.authenticate` returns a status of `Idx.Status:PENDING`. This status indicates that an additional factor needs to be verified before the user is successfully signed in. The response also includes a `nextStep` field that identifies the input parameters of the next step, in this case a phone authenticator key:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-sign-in-pwd-phone/main/nodejs/integrationsteps.md:13]:
```suggestion
If the username and password are valid, `idx.authenticate` returns a status of `Idx.Status:PENDING`. This status indicates that an additional factor needs to be verified before the user is successfully signed in. The response also includes a `nextStep` field that identifies the input parameters of the next step, which in this case is for a phone authenticator key:
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-sign-in-pwd-phone/main/nodejs/integrationsteps.md [hunk 3]
```diff
@@ -34,7 +34,7 @@ In this use case, only the **phone** factor is displayed in the list of authenti
 
 >**Note:** For the SDK sample application, each user must set up a phone number for phone verification to see this authenticator option in the page.
 
-When the user selects the **phone** factor, call `idx.authenticate` and pass in the authentication phone type, `{ authenticator: 'phone' }`. If the call is successful, the method returns a status of `Idx.Status:PENDING`, which indicates that the SDK needs a phone verification method in order to send the verification code. The `nextStep` field includes the input for this method.
+When the user selects the **phone** factor, call `idx.authenticate` and pass in the authentication phone key, `{ authenticator: AuthenticatorKey.OKTA_PHONE_NUMBER }`. If the call is successful, the method returns a status of `Idx.Status:PENDING`, which indicates that the SDK needs a phone verification method in order to send the verification code. The `nextStep` field includes the input for this method.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-sign-in-pwd-phone/main/nodejs/integrationsteps.md:37]:
```suggestion
When the user selects the **phone** factor, call `idx.authenticate` and pass in the authentication phone key, `{ authenticator: AuthenticatorKey.OKTA_PHONE_NUMBER }`. If the call is successful, the method returns a status of `Idx.Status:PENDING`, which indicates that the SDK needs a phone verification method to send the verification code. The `nextStep` field includes the input for this method.
```

**Review by susanharper-okta — APPROVED** [review]
couple nits - approving to not hold you up

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-self-reg/main/nodejs/integrationsteps.md [hunk 4]
```diff
@@ -46,8 +45,7 @@ If the org's application is properly configured with multiple factors, `idx.regi
       return true;
     case 'enroll-authenticator':
       redirect({ req, res, path: `/enroll-authenticator/${type}` });
-      return true;
-    case 'authenticator-enrollment-data':
+      return true;bd-okta-452597-Update-oie-sdk-nodejs-guide-authenticator-key    case 'authenticator-enrollment-data':
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-self-reg/main/nodejs/integrationsteps.md:48]: typo here?

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-self-reg/main/nodejs/integrationsteps.md [hunk 5]
```diff
@@ -129,7 +127,7 @@ If the request to skip the optional authenticator is successful, the SDK returns
 
 #### Option 2: The user selects the phone authenticator
 
-After the user selects the phone authenticator value, and `idx.register` is called with this value, the response returns a status of `Idx.Status:PENDING` and a `nextStep` field that requires phone registration data, which includes the phone number and verification method (SMS or voice). The user is directed to a page to enroll the phone data:
+After the user selects the phone authenticator value, and `idx.register` is called with this authenticator key (`Authenticator: AuthenticatorKey.OKTA_PHONE_NUMBER`), the response returns a status of `Idx.Status:PENDING` and a `nextStep` field that requires phone registration data, which includes the phone number and verification method (SMS or voice). The user is directed to a page to enroll the phone data:
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-self-reg/main/nodejs/integrationsteps.md:130]: OKTA_PHONE_NUMBER -> PHONE_NUMBER

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-sign-in-pwd-phone/main/nodejs/integrationsteps.md [hunk 6]
```diff
@@ -34,7 +34,7 @@ In this use case, only the **Phone** factor appears in the list of authenticator
 
 >**Note:** For the SDK sample application, each user must set up a phone number for phone verification to see this authenticator option in the page.
 
-When the user selects the **Phone** factor, call `idx.authenticate` and pass in the authentication phone type `{ authenticator: 'phone' }`. If the call is successful, the method returns a status of `Idx.Status:PENDING` that indicates that the SDK needs a phone verification method to send the verification code. The `nextStep` field includes the input for this method.
+When the user selects the **phone** factor, call `idx.authenticate` and pass in the authentication phone key, `{ authenticator: AuthenticatorKey.OKTA_PHONE_NUMBER }`. If the call is successful, the method returns a status of `Idx.Status:PENDING`, which indicates that the SDK needs a phone verification method to send the verification code. The `nextStep` field includes the input for this method.
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-sign-in-pwd-phone/main/nodejs/integrationsteps.md:37]: OKTA_PHONE_NUMBER -> PHONE_NUMBER

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-sign-in-pwd-phone/main/nodejs/integrationsteps.md [hunk 7]
```diff
@@ -57,7 +57,7 @@ Build the phone verification method entry page that accepts either SMS or voice
 
 When the user enters a phone verification method (either SMS or voice verification) and clicks **Next**, a call to `idx.authenticate` is made that passes in the following values:
 
-* Authenticator (type): `{ authenticator: 'phone' }`
+* Authenticator key: `{ authenticator: AuthenticatorKey.OKTA_PHONE }`
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-sign-in-pwd-phone/main/nodejs/integrationsteps.md:60]: OKTA_PHONE -> PHONE_NUMBER

**Review by shuowu-okta — APPROVED** [review]
key references can be found in https://github.com/okta/okta-auth-js/blob/master/lib/idx/types/index.ts#L41

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-self-reg/main/nodejs/integrationsteps.md (hunk 4 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-self-reg/main/nodejs/integrationsteps.md:48]: Yikes. Yes. Thanks for catching this.

**Pull request merged**